evm-tests/jsontests/src/state.rs

```diff
@@ -722,11 +727,175 @@ fn check_create_exit_reason(
 	false
 }
 
+/// Assert vicinity validation to ensure that test os expected validation error
+#[allow(clippy::cognitive_complexity)]
+fn assert_vicinity_validation(
+	reason: &InvalidTxReason,
+	states: &[PostStateResult],
+	spec: &ForkSpec,
+	name: &str,
+) {
+	match *spec {
+		ForkSpec::Istanbul => match reason {
+			InvalidTxReason::GasPriseEip1559 => {
+				for (i, state) in states.iter().enumerate() {
+					let expected = state
+						.expect_exception
+						.as_deref()
+						.expect("expected error message for test: [{spec}] {name}:{i}");
+					let is_checked = expected == "TR_TypeNotSupported";
+					assert!(
+						is_checked,
+						"unexpected error message {expected:?} for:  [{spec:?}] {name}:{i}",
+					);
+				}
+			}
+			_ => panic!("Unexpected validation reason: {reason:?} [{name}]"),
+		},
+		ForkSpec::Berlin => match reason {
```

Comment:
It looks like the branches `ForkSpec::Istanbul` and `ForkSpec::Berlin` could be combined:
```rust
		ForkSpec::Istanbul | ForkSpec::Berlin => match reason {
			InvalidTxReason::GasPriseEip1559 => {
				for (i, state) in states.iter().enumerate() {
					let expected = state
						.expect_exception
						.as_deref()
						.expect("expected error message for test: [{spec}] {name}:{i}");
					let is_checked = expected == "TR_TypeNotSupported";
					assert!(
						is_checked,
						"unexpected error message {expected:?} for:  [{spec:?}] {name}:{i}",
					);
				}
			}
			_ => panic!("Unexpected validation reason: {reason:?} [{name}]"),
		}
```

Reply:
Done: 09cdb6f9634d9b4b76ceaa2f1237c0cecebfbc4d